File: examples/13.delete-user-with-python/delete-user/handler.py
import boto3, json, os

client = boto3.resource('dynamodb')

IS_OFFLINE = os.getenv('IS_OFFLINE', False)
if IS_OFFLINE:
    boto3.Session(
        aws_access_key_id='ACCESS_KEY',
        aws_secret_access_key='SECRET_KEY',
    )
    client = boto3.resource('dynamodb', endpoint_url='http://localhost:8000')

table = client.Table('user-crud-sls-app')

def delete_user(event, context):
    user_id=event['pathParameters']['idx']
    result=table.delete_item(Key = {'pk': user_id})
    
    body = json.dumps({
        "message": f"user {user_id} deleted"
    })

    response = {
        "statusCode": result['ResponseMetadata']['HTTPStatusCode'],
        "body": body
    }
    return response

File: curso1/10.Recapitulacion-de-la-aplicacion-del-proyecto-y-creamos-nuestro-GET/handler.js
const AWS = require("aws-sdk")
const dynamoDB = new AWS.DynamoDB.DocumentClient()

const getUser = async (event, context) => {
    const params = {
        TableName: 'user-crud-sls-app',
        KeyConditionExpression: '#pk = :pk',
        ExpressionAttributeValues: { ':pk': '1' },
        ExpressionAttributeNames: { '#pk': 'pk' }
    }
    const user = await dynamoDB.query(params).promise()
    return {
        "statusCode": 200,
        "body": JSON.stringify(user)
    }
}

module.exports = {
    getUser
}
